verilog/tools/kythe/kythe_facts_extractor.h

```diff
@@ -57,13 +57,13 @@ class KytheFactsPrinter {
 
 std::ostream& operator<<(std::ostream&, const KytheFactsPrinter&);
 
-// Extracted Kythe indexing facts and edges.
-struct KytheIndexingData {
-  // Extracted Kythe indexing facts.
-  std::set<Fact> facts;
-
-  // Extracted Kythe edges.
-  std::set<Edge> edges;
+// Output sink interface for producing the Kythe output.
+class KytheOutput {
```

Comment:
Alternatively: you could formulate this as a pair functions, a FactHandler and an EdgeHandler, no need for any virtual function dispatch, perhaps simpler?  

Up to you, it's a matter of taste.

Reply:
I agree (and in fact my first implementation was like that), but I wanted to keep the change small in the first step (as there was already a streaming interface that needed extension).
More refactoring will happen in this file :)